python_modules/libraries/dagster-ge/dagster_ge/factory.py

```diff
@@ -72,11 +83,9 @@ def ge_validation_solid(context, pandas_df):
             )
             validation_operator = "ephemeral_validation"
         suite = data_context.get_expectation_suite(suite_name)
-        batch_kwargs = {
-            "dataset": pandas_df,
-            "datasource": datasource_name,
-        }
-        batch = data_context.get_batch(batch_kwargs, suite)
+        final_batch_kwargs = batch_kwargs or {"dataset": dataset}
+        final_batch_kwargs["datasource"] = datasource_name
```

Comment:
it may make sense to check for the presence of the key `'datasource'` in `batch_kwargs` and warn/log if it is present